src/Parsers/ASTQueryWithTableAndOutput.cpp

```diff
@@ -6,10 +7,75 @@
 namespace DB
 {
 
+String ASTQueryWithTableAndOutput::getDatabase() const
+{
+    String name;
+    tryGetIdentifierNameInto(database, name);
+    return name;
+}
+
+String ASTQueryWithTableAndOutput::getTable() const
+{
+    String name;
+    tryGetIdentifierNameInto(table, name);
+    return name;
+}
+
+void ASTQueryWithTableAndOutput::setDatabase(const String & name)
```

Comment:
Having several identical not short methods in two different places seems not ok, as fixing bug in one place someone will forget (or not even be aware of) that there is a second place.. Is it possible to make them in one place so that both classes could access it?